lib/OpenQA/Worker/Cache/Task/Asset.pm

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Task::Asset;
+
+use Mojo::Base 'Mojolicious::Plugin';
+use constant LOCK_RETRY_DELAY   => 30;
+use constant MINION_LOCK_EXPIRE => 999999999;
+
+use OpenQA::Worker::Cache;
+use OpenQA::Worker::Common;    # FIXME: This import needs to disappear.
+has ua => sub { Mojo::UserAgent->new };
+has cache => sub {
+    my ($worker_settings, undef) = OpenQA::Worker::Common::read_worker_config(undef, undef);
+    OpenQA::Worker::Cache->new(
+        host     => 'localhost',
+        location => ($ENV{CACHE_DIR} || $worker_settings->{CACHEDIRECTORY}));
+};
+
+#has token => sub { int(rand(999999999999)) };
+
+sub token {
+    shift->ua->get('http://localhost:3000/session_token')->result->json->{session_token};
+}
+
+sub dequeue {
+    !!(
+        shift->ua->get('http://localhost:3000/dequeue/' . shift)->result->json->{status} eq
```

Comment:
I guess eventually we'd need to write down the ports that we're using... somewhere... but picking a different port than 3000 would be nice... :)

Reply:
Yeah, i'm aware - still not reached that point